lib/rust/parser/src/syntax/tree.rs

```diff
@@ -206,6 +206,8 @@ macro_rules! with_ast_definition { ($f:ident ($($args:tt)*)) => { $f! { $($args)
         },
         /// A variable assignment, like `foo = bar 23`.
         Assignment {
+            /// Documentation applied to the variable.
+            pub doc_line: Option<DocLine<'s>>,
```

Comment:
I don't see any added field for `TypeDef` - don't these have documentation too?

Reply:
The statements the frontend needs to be able to document (Function, Assignment, ExpressionStatement) have been ported to the compound representation. Some other statement types are documented by a preceding free `Documentation` line in the block, which is currently easiest for the backend. The remaining types should be converted along with improvements to the IR types to eliminate its use of independent documentation nodes (and other modifier-line types).